publishing.md

```diff
@@ -64,16 +64,48 @@ should set this value to prevent accidental pushes to rubygems.org:
 See the [Resources](/resources) guide for an up-to-date listing of options for
 private gem servers.
 
+Authenticating for Publishing
+-----------------------------
+
+The push command will prompt for authentication, and this is the basic workflow
+for authenticating with [RubyGems.org][rubygems-url].
+
+However, if your `gem push` is
+automated, or you'd prefer to authenticate prior to pushing, the `gem signin`
+command provides the ability to store authentication information.
+
+To begin, you'll need to create an account on RubyGems.org. Visit the
+[sign up][rubygems-signup-url] page and supply an email address that you control,
+a handle (username), and a password.
+
+After creating the account, use your email and password when signing in.
+(RubyGems saves the credentials in ~/.gem/credentials for you so you only need
+to log in once.)
+
+Alternatively, you can authenticate with a private registry by providing the `--host`
+flag while signing in.
+
+To authenticate and enable you to push version 0.1.0 of a new gem named 'squid-utils':
+
+    $ gem signin squid-utils-0.1.0.gem
```

Comment:
With current signin command implementation, I don't think it requires
or accept a gem as argument, so I think it should just be:

    gem signin